amethyst_network/src/connection/socket_state.rs

```diff
@@ -0,0 +1,43 @@
+use amethyst_core::specs::{Component, VecStorage};
+
+use std::collections::{HashMap};
+use std::collections::hash_map::Entry;
+use std::net::SocketAddr;
+use std::sync::{Arc, Mutex};
+
+use packet::{Packet, CompletePacket};
+use net::{NetIdentity, NetConnection};
+use uuid::Uuid;
+use itertools::Itertools;
+use std::borrow::BorrowMut;
+
+pub struct SocketState<E: 'static + Send + Sync> {
+    connections: HashMap<SocketAddr, NetConnection<E>>
+}
+
+impl<E> SocketState<E>  where E: 'static + Send + Sync {
+    pub fn new() -> SocketState<E> {
+        SocketState { connections: HashMap::new() }
+    }
+
+    pub fn preprocess_packet(&mut self, p: Packet) -> (SocketAddr, Vec<u8>) {
+        let connection = self.connections.entry(p.addr).or_insert(NetConnection::new(p.addr));
+        connection.waiting_packets.enqueue(connection.seq_num, p.clone());
+        let final_packet = ::utils::assemble_packet(connection.seq_num, p.clone(), connection);
```

Comment:
I would rather not use this syntax, just `use utils::assemble_packet`.

Reply:
I see you have created a separate repo for the UPD part. You want me to finish this PR? There are still things we need to keep from this PR thought! Like in the system when the packet gets assembled for example. See https://github.com/LucioFranco/amethyst-protocol/pull/1 this where I have continued working on where you left off.